File: DSA/DSA06014 TỔNG CẶP SỐ NGUYÊN TỐ.cpp
#include <bits/stdc++.h>
using namespace std;

vector<bool> prepare()
{
    vector<bool> prime(1e6+1, true);
    prime[0] = prime[1] = false;
    for (int i = 2; i*i < 1e6+1; i++)
    {
        for (int j = i*i; j < 1e6+1; j+=i)
        {
            prime[j] = false;
        }
    }
    return prime;
}

void submit(vector<bool> &prime)
{
    int n;
    cin >> n;
    
    bool check = false;
    for (int i = 0; i < n; i++)
    {
        if (prime[n-i] && prime[i])
        {
            check = true;
            cout << i << " " << n-i;
            break;
        }
    }

    if (!check) cout << -1;
    

    cout << endl;
}


int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    vector<bool> prime = prepare();
    while (so_bo_test--)
    {

        submit(prime);
    }

    return 0;
}

File: DSA_python/DSA06014 TỔNG CẶP SỐ NGUYÊN TỐ.py
def prepare():
    prime = [True]*(10**6+1)
    prime[0] = prime[1] = False
    for i in range(2, 10**6+1):
        for j in range(i*i, 10**6+1, i):
            prime[j] = False
    return prime

def submit():
    n = int(input())
    check = False
    for i in range(n):
        if prime[n-i] == True and prime[i] == True:
            check = True
            print(i, n-i)
            break
    
    if check == False: print(-1)


test = int(input())
prime = prepare()
for i in range(test):
    submit()
